### DISCUSSION TITLE
oidc_unsolicited_proto_state: could not parse JWT from state: invalid unsolicited response

### DISCUSSION THREAD

**Discussion by djordje-petrovic** [OpenIDC/mod_auth_openidc#500] [Q&A]
Hi,

Here's the config:

    _OIDCProviderMetadataURL https://example.com/.well-known/openid-configuration
    OIDCClientID <redacted>
    OIDCClientSecret <redacted>
    OIDCCacheShmEntrySizeMax 120000
    OIDCSessionInactivityTimeout 36000
    OIDCStateTimeout 900
    OIDCCookieDomain app.example.com
    OIDCScope "openid profile name email"
    OIDCRedirectURI https://app.example.com/secure/redirect_uri/
    OIDCRedirectURLsAllowed ^https://apps.example1.com$
    OIDCDefaultURL https://app.example.com/gallery/
    OIDCHTMLErrorTemplate /var/www/html/custom_errors/mod_oidc_error.html
    OIDCCryptoPassphrase secret
    OIDCPassClaimsAs environment
    OIDCStateMaxNumberOfCookies 2 true
    OIDCCacheType shm
    OIDCCacheShmMax 1500_

And the error during login process which ends with saml loop:

_[Thu Dec 10 10:37:10.105931 2020] [auth_openidc:error] [pid 20662:tid 140547856869120] [client 10.135.7.100:48830] oidc_restore_proto_state: no "mod_auth_openidc_state_mgCtI0O5RlyYKJidxfLgOG9QA2c" state cookie found
[Thu Dec 10 10:37:10.105993 2020] [auth_openidc:error] [pid 20662:tid 140547856869120] [client 10.135.7.100:48830] oidc_unsolicited_proto_state: could not parse JWT from state: invalid unsolicited response: [src/jose.c:808: oidc_jwt_parse]: cjose_jws_import failed: invalid argument [file: jws.c, function: cjose_jws_import, line: 781]
[Thu Dec 10 10:37:10.105996 2020] [auth_openidc:error] [pid 20662:tid 140547856869120] [client 10.135.7.100:48830] oidc_authorization_response_match_state: unable to restore state_

I've looked through the old similar issues (notably this one https://github.com/zmartzone/mod_auth_openidc/issues/213) and the solutions in those don't work for us.
So any clues?

**Comment by zandbelt** [comment]
that's probably an issue with the Chrome browser and samesite cookies

**Comment by djordje-petrovic** [comment]
It happens in all browsers.
Also this is response header for set-cookie:

set-cookie: mod_auth_openidc_session=98aa2897-7fdb-4099-aa33-5436cb488821; Path=/; Domain=example.com; Secure; HttpOnly; SameSite=None

Any ideas?

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by zandbelt** [comment]
that's a set-cookie header for the session which seems to be correct; you mentioned a problem with the state cookie earlier, now I'm confused about what is wrong

**Reply by ganesh-bh** [reply]
@djordje-petrovic  Did the issue got resolved?
I am facing similar issue in my application please help me
